src/data/list/min_max.lean

```diff
@@ -315,6 +315,22 @@ begin
   { simpa [h y (mem_cons_self _ _)] using IH (λ x hx, h x $ mem_cons_of_mem _ hx) }
 end
 
+lemma le_max_of_exists_le (l : list α) (a : α) {x : α} (hx : x ∈ l) (h : a ≤ x) :
+  a ≤ l.foldr max ⊥ :=
+begin
+  induction l with y l IH,
+  { exact absurd hx (not_mem_nil _), },
+  { by_cases hl_nil : l = nil,
+    { rw [hl_nil, mem_singleton] at hx,
+      rw hx at h,
+      simp only [foldr, foldr_cons, le_max_of_le_left h] },
+    { cases hx with hy hl,
+      simp only [foldr, foldr_cons],
+      { rw hy at h,
+        exact le_max_of_le_left h, },
+      { exact le_max_of_le_right (IH hl) }}}
```

Comment:
This `by_cases` is pointless, the second case doesn't use `hl_nil`:

```suggestion
  { cases hx with hy hl,
    simp only [foldr, foldr_cons],
    { rw hy at h,
      exact le_max_of_le_left h, },
    { exact le_max_of_le_right (IH hl) } }
```

Reply:
You can also golf the `cases` to `obtain rfl | hl := hx`

Reply:
Shouldn't it be this, actually?
```suggestion
  { cases hx with hy hl,
    { simp only [foldr, foldr_cons],
      rw hy at h,
      exact le_max_of_le_left h, },
    { exact le_max_of_le_right (IH hl) } }
```

Reply:
Thanks!